tests/func/experiments/test_experiments.py

```diff
@@ -94,27 +94,45 @@ def test_update_with_pull(tmp_dir, scm, dvc, mocker):
 
 
 @pytest.mark.parametrize(
-    "change, expected",
+    "changes, expected",
     [
-        ["foo.1.baz=3", "foo: [bar: 1, baz: 3]"],
-        ["foo.0=bar", "foo: [bar, baz: 2]"],
-        ["foo.1=- baz\n- goo", "foo: [bar: 1, [baz, goo]]"],
+        [["foo=baz"], "{foo: baz, goo: {bag: 3}, lorem: false}"],
+        [["foo=baz,goo=bar"], "{foo: baz, goo: bar, lorem: false}"],
+        [
+            ["goo.bag=4"],
+            "{foo: [bar: 1, baz: 2], goo: {bag: 4}, lorem: false}",
+        ],
+        [["foo[0]=bar"], "{foo: [bar, baz: 2], goo: {bag: 3}, lorem: false}"],
+        [
+            ["foo[1].baz=3"],
+            "{foo: [bar: 1, baz: 3], goo: {bag: 3}, lorem: false}",
+        ],
+        [
+            ["foo[1]=- baz\n- goo"],
```

Comment:
This syntax is not easy to understand/use. At a first glance, I associated those `=-` together and assumed it was removing items from the list.

Ideally, I'd like to have the following syntax:
```console
$ dvc exp run --params "foo[1]=[baz, goo]" "bar={a:2, b: 3}"
```

I understand that we split it up by comma and achieve that. We should consider using a different delimiter, maybe even space and make it look json-like.

Reply:
It seems a natural progression to support appending and removing similar to [Hydra's syntax](https://hydra.cc/docs/next/advanced/override_grammar/basic/) (they use `~foo` to remove and `+foo=value` to append). 

Or, maybe even use Hydra later directly.

Reply:
Note that I didn't change anything regarding the syntax. This is the syntax already accepted by the yaml parser being used, which is separate from this change. I'm simply documenting what currently works.

That said, I agree that it it would be nice to be able to specify arguments that include commas, and I think that your suggestion of simply specifying a list instead of a single string would be the most straight-forward way of doing it.